### PR TITLE
44 misc thread safety use case architecture

### PR DESCRIPTION
**Pull request by Serdiuk-Andrii**

### CODE REVIEW

File: src/main/java/usecases/disconnecting/DcInteractor.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    private final LobbyManager lm;
+    private final DcOutputBoundary dcOutputBoundary;
+    private final Lock playerPoolLock;
+    private final Lock gameLock;
+
+    /**
+     * Constructor for DcInteractor
+     * @param lm Lobby Manager
+     * @param dcOutputBoundary DcOutputBoundary
+     * @param playerPoolLock The lock used for synchronization with other object that access the player pool
```
Comment (tostistos) [inline src/main/java/usecases/disconnecting/DcInteractor.java:25]: Missing @param comment for gameLock

**Review by tostistos — COMMENTED** [review]
Looks good, but missing a @param comment for gameLock (see comment).

**Review by mariana0412 — DISMISSED** [review]
Good decisions!

**Comment by Serdiuk-Andrii** [comment]
> Looks good, but missing a @param comment for gameLock (see comment).

Thank you, it has been fixed

**Review by cotopaxi64 — APPROVED** [review]
Ok, good tweaking.

**Review by AlekseyPanas — DISMISSED** [review]
To my eye, this lock architecture is good and should avoid all thread safety issues

**Comment by Serdiuk-Andrii** [comment]
> Ok, good tweaking.

=> Implementing a complex lock system ≃ tweaking :)

**Review by bac423 — APPROVED** [review]
Great. Documentation approved

**Review by cotopaxi64 — APPROVED** [review]
Approved.

**Pull request merged**